---
title: "AMPLseq Tutorial"
author: "Laboratory of Genomic and Evolutionary Biology of Malaria Parasites"
date: "2022-09-27"
output:
  html_document:
    toc: true
    number_sections: true
    toc_float: true
    code_folding: show
editor_options: 
  markdown: 
    wrap: 72
  chunk_output_type: inline
---

The purpose of this tutorial is to introduce researchers at the Laboratory of Genomic and Evolutionary Biology of Malaria Parasites at Harvard university in the analysis of genotyping information generated through amplicon sequencing (also called Microhaplotype genotyping). This tutorial will cover:

1.  Importing and handling tables in CIGAR format in R environment.

2.  Adding or generating metadata.

3.  Performance of the genotyping process.

4.  Molecular surveillance of drug resistance.

5.  Monitoring transmission intensity.

6.  Measuring geographic connectivity.

In this tutorial we will use a Colombian data set as a example. These samples comes from a collaboration with the group of Caucaseco, which has collected samples of *P. falciparum* from 2019 to 2022 in three municipalities located in the pacific coast of Colombia.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# Importing and handling tables in CIGAR format in R environment

Our first step will be to call all required packages and functions in the R environment:

```{r 'Loading required packages and functions'}
source('amplseq_required_libraries.R')
source('amplseq_functions.R')
```

As we mentioned earlier, this tutorial start with .tsv table in CIGAR format that has been generated with a previous pipeline created in the lab. The output of this pipeline has the following structure:

$\begin{array}{c|c:c:c:c:c:c} \text{sampleID}&Gene_1,Allele_1&Gene_1,Allele_2&...&Gene_1,Allele_k&... & Gene_m, Allele_{k_m}\\ \hline ID_1 & \text{Read counts} &&& \\ \hdashline ... &&&&\\ \hdashline ID_n &&&& \end{array}$

Where the $Allele$ is coded in CIGAR format, typing "." for the reference allele, $[0-9]*[A,T,C,G]$ for each point mutation observed, and $[0-9]*[D,I]=[A,T,C,G]*$ for indels (Insertion and Deletions).

These CIGAR tables are stored in the server of the lab using the following path structure:

`"STUDYNAME/RUNNAME/dada2/run_dada2/CIGARVariants.out.tsv"`

As this structure is maintained across all studies in the lab, we have created a function called `fx_read_cigar_tables` that only requires two arguments: `paths` (name of the folder of the study or the `STUDYNAME`), and `sample_id_pattern` (a pattern string in the IDs of the samples that differentiate them from controls). For this particular example, the path will be `"colombia/"` , while the pattern will be `"SP"`.

```{r 'Uploading data sets'}
cigar_object = read_cigar_tables(paths = "colombia/", sample_id_pattern = "SP")
```

This function generates a list containing two data.frames, 1) a `cigar_table`, where genotyping information is stored in cigar format; and 2) a `metadata` table with 3 variables (columns), `samples` which contains samples Id's, `run` which contains the run number (useful for comparing performance across batches of samples), and `typeofSamp` which differentiates between controls and samples of interest. In order to view these tables use `View(cigar_table$cigar_table)` or `View(cigar_table$metadata)` in the console.

# Adding Metadata

Metadata would come from different sources, sometimes the sample ID incorporates metadata in their coding system, other times metadata can be stored in a different table, having one column specifying the sample ID. In the case of the Colombian data set, we will perform the analysis at the level of Municipality and in intervals of three months starting from the first day of collection of samples. The information of sampling location of the samples is incorporated in the `sample_id code`, while the time of collection is in a separate table. Thus for sampling location all codes starting with **"SP0101254"** come from **"Buenaventura"**, samples starting with **"SP0112286"** come from **"Quibdo"**, and samples starting with **"SP0126288"** come from **"Guapi"**. In this data set, there is one sample that has an spelling error in its code, the code is `"SP001254204"`, and we are going to replace it with `"SP012542004"`. Then, to define sampling location we are going to create a new variable in our `metadata` table called `Population` using the function `mutate`.

```{r 'Adding Population'}
# Upload metadata from an external source
metadata = read.csv('Gates_Colombia_metadata.csv')

# Correcting sample "SP001254204"----
cigar_object@metadata[cigar_object@metadata[["Sample_id"]] == "SP001254204",][["Sample_id"]] = "SP0101254204"
rownames(cigar_object@cigar_table) = cigar_object@metadata[["Sample_id"]]

# Merge the external metadata with our cigar_object
cigar_object@metadata = merge(cigar_object@metadata,
                              metadata,
                              by = 'Sample_id',
                              all.x = T)

```

Moreover, there are some artifacts that `dada2` pipeline doesn't resolve well like the generation of chimeras (artificial fusion of two different haplotypes during the PCR amplification) in polyclonal samples. That is the case of the alleles `PF3D7_1302900,1G` and `PF3D7_0612900,215A` that are observed in polyclonal infections only. So we are going to proceed to remove both alleles from our `cigar_table`. Additionally, `PvDHFR` locus was amplified in order to detect mixed infections with *Plasmodium vivax*, so we are going to remove this locus too.

```{r 'Removing artifacts and Pvivax control locus'}

# removing allele PF3D7_1302900,1G----
# this polymorphism is always present as a minor allele in polygenomic samples, increasing the proportion of polygenomic infections to up to 27%

cigar_object@cigar_table = cigar_object@cigar_table[,!grepl("PF3D7_1302900,1G", colnames(cigar_object@cigar_table))]
cigar_object@cigar_table = cigar_object@cigar_table[,!grepl("PF3D7_0612900,215A", colnames(cigar_object@cigar_table))]

# removing locus PvDHFR----
# this loci belongs to P. vivax

cigar_object@cigar_table = cigar_object@cigar_table[,!grepl("PvDHFR", colnames(cigar_object@cigar_table))]

```

Then we will upload a metadata table from an external source. For this purpose we are going to import the file called `metadata.csv` into our R environment. From this table we want to extract the date of beginning of symptoms (`date_of_symp`) to include it in our analysis. Unfortunately, not all sampled individuals has answered that question in the survey, for that reason we are going to impute that information using the date of collection of the sample. Then, we are going to collapse information at the level of month (`month_of_symp`), and we are going to merge our current `metadata` with the `metadata` from the external source. Finally, we are going to impute missing month dates using the month date from a sample located one row above (sample collected just before the sample with missing information).

```{r 'Ploting the number of collected samples by Municipality over time', echo=FALSE}

# Define the temporal unit of analysis
dates = sort(unique(cigar_object@metadata$Month_of_Collection))

# Define the geographic unite of analysis
pop_levels = unique(cigar_object@metadata$Subnational_level2)
pop_levels = pop_levels[!is.na(pop_levels)]
pop_levels = pop_levels[c(2, 1, 3, 4, 5)]
pop_colors = c("firebrick3", "dodgerblue3", "gold3", "darkseagreen3", "lightsalmon2")

plot_temporal_collection_of_samples = cigar_object@metadata %>%
  filter(!is.na(Subnational_level2)) %>%
  group_by(Subnational_level2, Month_of_Collection) %>%
  summarise(nsamples= n())%>%
  ggplot(aes(x = Month_of_Collection, y = nsamples, fill = factor(Subnational_level2,
                                                              levels = pop_levels)))+
  geom_col()+
  theme_bw()+
  scale_fill_manual(values = pop_colors)+
  facet_wrap(.~factor(Subnational_level2,
                      levels = pop_levels),
             strip.position = "top", ncol = 1)+
  labs(title = 'B) # Samples collected over time',
       y = "Collected samples by PCD",
       x = "Months")+
  theme(axis.text = element_text(size = 12),
        axis.text.x = element_text(angle = 315, vjust = 0),
        axis.title = element_text(size = 12),
        strip.text = element_text(size = 12),
        legend.position =  "none")+
  scale_alpha_manual(values = c(1,.6))+
  scale_x_discrete(limits = dates)
```

```{r 'Map of study sites', echo=FALSE}

# Get all shape files from LAC countries at national level
# SpatialPolygonsDataFrame at country level (level = 0)

col0.spldf = gadm_sp_loadCountries("COL",level = 0, basefile = "./GADMTools/world.rds.files/rds0/")

# SpatialPolygonsDataFrame at Municipality level (level = 2)
col2.spldf = gadm_sp_loadCountries("COL",level = 2, basefile = "./GADMTools/world.rds.files/rds2/")

# Label for Municipalities
localities = data.frame(Localities = factor(pop_levels,
                                            levels = pop_levels[5:1]),
                        long = -70,
                        lat = c(6, 8, 10, 12, 14),
                        colors = pop_colors[5:1])

# Plot the map

## Polygon dot plot of observed G6PDd frequency by LAC country----
plot_study_sites = ggplot() +
  geom_polygon(data=col0.spldf$spdf, aes(x=long, y=lat, group = group), fill = "floralwhite", color = "gray40", linewidth=.3) +
  geom_polygon(data=col2.spldf[["spdf"]][col2.spldf[["spdf"]][["NAME_2"]] == "Quibdó",], aes(x=long, y=lat, group = group), fill = "firebrick3", color = "gray40", linewidth=.3) +
  geom_polygon(data=col2.spldf[["spdf"]][col2.spldf[["spdf"]][["NAME_2"]] == "Buenaventura",], aes(x=long, y=lat, group = group), fill = "dodgerblue3", color = "gray40", linewidth=.3) +
  geom_polygon(data=col2.spldf[["spdf"]][col2.spldf[["spdf"]][["NAME_2"]] == "Guapí",], aes(x=long, y=lat, group = group), fill = "gold3", color = "gray40", linewidth=.3) +
  geom_polygon(data=col2.spldf[["spdf"]][col2.spldf[["spdf"]][["NAME_2"]] == "Tumaco",], aes(x=long, y=lat, group = group), fill = "darkseagreen3", color = "gray40", linewidth=.3) +
  geom_polygon(data=col2.spldf[["spdf"]][col2.spldf[["spdf"]][["NAME_2"]] == "Puerto Inírida",], aes(x=long, y=lat, group = group), fill = "lightsalmon2", color = "gray40", linewidth=.3) +
  #geom_point(data=localities, aes(x=long, y=lat, color=Localities), size = 5) +
  #geom_label(data=localities, aes(x=long + c(1.3, 2.3, 1.5, 1.2), y=lat, label = Localities)) +
  theme_bw()+
  labs(title = "A) Study sites in Colombia",
       x = "Longitude",
       y = "Latitude")+
  scale_x_continuous(limits = c(-80, -65))+
  scale_y_continuous(limits = c(-5, 15))+
  scale_color_manual(values = localities$colors)+
  theme(legend.position = "none",
        text = element_text(size = 10),
        axis.text = element_text(size = 10),
        title = element_text(size = 10))+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 12))

```

```{r 'Plot of number of collected samples by Municipality over time whit map', echo=FALSE, warning=FALSE, message=FALSE, fig.width=10, fig.height=5, fig.cap='**Figure 1:** Number of collected samples by Municipality over time'}
plot_study_areas = ggdraw()+
  draw_plot(plot_study_sites,
            x = 0,
            y = 0,
            width = .5,
            height = 1)+
  draw_plot(plot_temporal_collection_of_samples,
            x = .5,
            y = 0,
            width = .5,
            height = 1)

plot_study_areas
```

# Performance of the genotyping process

A first step in analyzing the genotypic data is to measure the performance of the genotyping process in terms of the proportion of samples amplified by each locus (amplification rate by locus) and the proportion of loci amplified per each sample (amplification rate per sample). In order to perform that task, we are going to convert our cigar_table to a more manageable format called loci_abd format, where amplicons or genes are going to be in columns, samples in rows, and the allele in cigar format is going to be in each cell of the matrix together with the read count of that allele in the sample.

$\begin{array}{c|c:c:c:c} \text{sampleID}&Gene_1&Gene_2&...& Gene_m\\ \hline ID_1 & Allele_{i_{i \in Alleles_{Gene_1}}}\text{:Read count} &&& \\ \hdashline ... &&&&\\ \hdashline ID_n &&&& \end{array}$

For that purpose we are going to use the function `fx_cigar2loci_abd.R`. This function requires 4 arguments. First a `cigar_object` (containing the cigar_table and the metadata). Then `markers`, which is a table with the set of markers and their names (in a column named `amplicon`). This argument is optional, and in case is `NULL`, the name of the amplicons will be extracted from the columns in the `cigar_object`, but information regarding chormosome location and length of the amplicon will be missing. The next arguments are `min_abd` and `min_ratio`, which are the minimum abundance (minimum read count) required to call an allele and the minimum ratio between the minor and major alleles in a polyclonal sample. By default these two arguments are 10 and 0.1 respectively. This function automatically differentiate between controls and samples of interest, and its output is a `ampseq_object` list with 6 slots: `loci_abd_table` where the genotypic information of the samples of interest is stored in a loci_abd format; the `metadata` of the samples of interest; `controls` with all the information of the controls; `markers` containing the information of the markers such as the chromosome location; and two empty slots required for the next steps which are `loci_performance` and `pop_summary`.

```{r "Generating a loci-abundance table"}
# (alleles and abundance will be in cells)
markers = read.csv("markers.csv")
ampseq = cigar2ampseq(cigar_object, markers = markers, min_abd = 10, min_ratio = .1, remove_controls = T)
```

## Amplification rate by locus

Once the data has been converted to the loci_abd format, we can use the function `fx_loci_amplification_rate` to measure the proportion of samples that have been amplified by each amplicon marker. This function requires two arguments, the `ampseq_object` and `threshold`, which defines the minimum proportion of samples that a marker should amplify in order to be keep it for further analysis (default 0.65). All markers below this `threshold` will be discarded.

```{r 'Filtering loci of low amplification rate'}
ampseq = locus_amplification_rate(ampseq, threshold = .65)
```

Thus in this data set `r ncol(ampseq$loci_abd_table)` loci had an amplification rate above 0.65, and `r ncol(ampseq$discarded_loci$loci_abd_table)` loci were discarded. All discarded loci are stored in the slot `discarded_loci` within the `ampseq_object`. The discarded loci were: `r paste(colnames(ampseq$discarded_loci$loci_abd_table), collapse = ', ')`.

```{r 'Ploting loci performance', fig.width=7, fig.height=7, fig.cap= '**Figure 2:** Proportion of amplified samples per locus. Top figure shows the distribution of the amplification rate (or proportion of amplified samples) of loci, with the number of loci in the y-axis and the amplification rate in the x-axis. Bottom figure shows the the chromosome location of each locus (chromosomes in y-axis and position in the x-axis) and the gradient color represents its amplification rate.'}
# Plot Amplification rate per loci ----

plot_locus_amplificatin_rate = ggdraw()+
  draw_plot(ampseq@plots$amplification_rate_per_locus+
              theme(axis.text = element_text(size = 12),
                    axis.title = element_text(size = 12),
                    legend.text = element_text(size = 12)),
            x = 0,
            y = 0,
            width = 1,
            height = .5)+
  draw_plot(ampseq@plots$all_loci_amplification_rate+
              theme(axis.text = element_text(size = 12),
                    axis.title = element_text(size = 12)),
            x = 0,
            y = .5,
            width = 1,
            height = .5)

plot_locus_amplificatin_rate

```

## Amplification rate by sample

The next step is to measure the proportion of loci amplified per each sample, also called the amplification rate of the samples. For that purpose we are going to use the function `fx_sample_amplification_rate` that also requires only two arguments, the `ampseq_object` and a `threshold`, which is the minmum proportion of loci that a sample should amplified (by default 0.8).

```{r 'Filtering samples of low amplification rate', fig.width=5, fig.height=5, fig.cap='**Figure 3:** Proportion of amplified loci per sample. The figure shows the distribution of the amplification rate (or proportion of amplified loci) of the samples, with the number of samples in the y-axis and the amplification rate in the x-axis.'}

ampseq = sample_amplification_rate(ampseq)

ampseq@plots$samples_amplification_rate+
         theme(axis.text = element_text(size = 12),
               axis.title = element_text(size = 12))

```

# Molecular surveillance of drug resistance

Surveillance of polymorphisms or allelic variants associated with resistance against antimalarials is of the utmost importance in public health. That is why our panel of markers includes 5 genes (9 markers in total, some genes are genotyped by more than one marker) that have polymorphisms associated with antimalarial resistance. In this section we will identify the different haplotypes of these genes present in our data set, and we will determine the frequency of each haplotype in each study area and in each quarter of the year. For this we will use the function `fx_drug_resistant_haplotypes` which has 7 arguments. The first argument is the `ampseq_object`. The second is a table with the reference alleles for each gene, and the polymorphisms associated with resistance. The structure of this table is as follows:

$$\begin{array}{c|c:c:c} \text{Chromosome} & \text{Gene_ID} & \text{Description} & \text{Mutation} & \text{Anotation} \\ \text{Pf3D7_04_v3} & \text{PF3D7_0417200} & dhfr & \text{N51I} & \text{Pyrimethamine resistance} \\ ... \end{array}$$

Where the mutation describes the reference amino acid (sensitive to the drug), the position in the amino acid chain, and finally the amino acid associated with resistance.

The third and fourth arguments refer to the common name of the gene (or the name by which it is identified in the `markers` table in the `ampseq_object`) and the gene ID in the gff file of the referential strain (in this case strain 3D7). The fifth and sixth arguments are the .gff and .fasta files of the genome of the reference strain, and the last argument, `variables`, is a vector that has the columns of the `metadata` table that will be used to define the subpopulations and generate the report graph. For this particular example we will define as `variables = c('samples', 'Population', 'quarter_of_collection')`, where `samples`, is the column that we will use to map the data, `Population` is the column to define the area of study, and `quarter_of_collection` defines the time scale.

The function generates a list with 3 tables and a graph. The first table contains the haplotypes found in each sample (rows) for each gene (columns). Each number in the haplotype indicates a position in the amino acid chain, the letter before the number indicates the reference allele (with sensitive phenotype), and the letter after the number is the allele observed in the sample. If the letters are in capital letters, that allele in that position has been previously reported as resistant, whereas the letters in lower case indicate that it is the first time that said position has been reported as polymorphic. In case the sample is polyclonal, each allele observed is separated by the following symbol `|`, and the order in which they are reported is based on their read count. Null data, that is, sections of the gene that have not been amplified, are completed with the question mark `?`.

The second table contains the inferred phenotype for each sample (rows) for each gene (columns). The third table contains the count of each haplotype in the population (number of times observed) and its frequency in the population (defined based on geographic and temporal information). It is important to note that polyclonal samples are included in this calculation, so the sum of frequencies may be greater than unity in some cases.

```{r 'Molecular surveillance of drug resistance', fig.width=12, fig.height=7, fig.cap='**Figure 4:** Frequency of haplotypes of genes associated with drug resistance. y-axis shows the frequency in the sub-population, x-axis shows the quarter of the year where the sample was collected, rows panels corresponds to the three study areas, and column panels represents each genotyped gene. Colors were assigned based on the possible phenotype, dark blue indicates sensitive, while dark red indicates resistance. Light colors where assigned for incomplete information'}

ampseq_drug = ampseq

ampseq_drug@gt = cbind(ampseq_drug@gt,
                                   ampseq_drug@discarded_loci$gt[
                                     rownames(ampseq_drug@discarded_loci$gt) %in%
                                                     rownames(ampseq_drug@gt),
                                     grepl(('PfDHFR|PfMDR1|PfDHPS|PfKelch13|PF3D7_1447900'),
                                           colnames(ampseq_drug@discarded_loci$gt))]
                                          )

ampseq_drug@markers = rbind(ampseq_drug@markers,
                            ampseq_drug@discarded_loci$markers[
                              grepl(('PfDHFR|PfMDR1|PfDHPS|PfKelch13|PF3D7_1447900'),
                                    ampseq_drug@discarded_loci$markers$amplicon),])

drug_resistant_haplotypes = drug_resistant_haplotypes(ampseq_drug,
                                                         reference_alleles = 'drugR_alleles.csv',
                                        gene_names = c('PfDHFR',
                                                       'PfMDR1',
                                                       'PfDHPS',
                                                       'PfKelch13',
                                                       'PF3D7_1447900'),
                                        gene_ids = c('PF3D7_0417200',
                                                     'PF3D7_0523000',
                                                     'PF3D7_0810800',
                                                     'PF3D7_1343700',
                                                     'PF3D7_1447900'),
                                        gff_file = "reference/3D7/PlasmoDB-59_Pfalciparum3D7.gff",
                                        fasta_file = "reference/3D7/PlasmoDB-59_Pfalciparum3D7_Genome.fasta",
                                        variables = c('Sample_id', 'Subnational_level2', 'Quarter_of_Collection'),
                                        na.var.rm = TRUE,
                                        filters = c('Subnational_level2;Quibdó,Buenaventura,Guapi,Tumaco,Puerto Inírida',                               'Quarter_of_Collection;2020-Q4,2021-Q1,2021-Q2,2021-Q3,2021-Q4,2022-Q1,2022-Q2,2022-Q3'))

```

For the PfMDR1 gene, the predominant haplotype was 86N184F1042d containing the Y184F mutation associated with resistance to lumefantrine and mefloquine. In the PfMDR2 gene (PF3D7_1447900) the most prevalent haplotype was the sensitive haplotype 479y 484T 494i. For the PfKelch13 gene, all the samples had the artemisinin-sensitive haplotype. In the PfDHFR gene, the most prevalent haplotype was 16a51I59C108N164I, which contains two mutations associated with resistance to pyremethamine (N51I and S108N). However, in Buenaventura and Guapi the presence of the sensitive haplotype for this drug was observed. In contrast, the PfDHPS gene haplotype, 436S437G540K581A613A resistant to sulfodoxine, was the most prevalent only in Quibdo, while in the other two study areas the most prevalent haplotype was the sensitive haplotype.

```{r 'Plot drug resistance', fig.width=12, fig.height=7, fig.cap='**Figure 4:** Frequency of haplotypes of genes associated with drug resistance. y-axis shows the frequency in the sub-population, x-axis shows the quarter of the year where the sample was collected, rows panels corresponds to the three study areas, and column panels represents each genotyped gene. Colors were assigned based on the possible phenotype, dark blue indicates sensitive, while dark red indicates resistance. Light colors where assigned for incomplete information'}

blues = brewer.pal(9, 'Blues')
reds = brewer.pal(9, 'Reds')

drug_resistant_haplotypes$haplo_freq_plot+
  theme(axis.text.x = element_text(angle = 90))+
  scale_fill_manual(values = c(
    'gray75',blues[6], reds[3], # MDR2
    reds[c(8,7)], blues[3], reds[6], blues[6], reds[c(9,3)], # DFHR
    'gray75',blues[4], reds[5], blues[c(2,6)], reds[c(4,9,6,8)], # DHPS
    blues[6], #K13
    'gray75', blues[c(4,2,3)], reds[c(4,5,7,9)], blues[2], reds[4], blues[c(3,4)], reds[c(5,6,7,9)] # MDR1
  ))+
  labs(y = 'Haplotype prevalence in infected individuals')

```


# Monitoring transmission intensity

The transmission of Malaria has an effect on the evolutionary processes that affect the parasite population. Thus, various genetic indicators are used to identify changes in the intensity of transmission at a spatial and temporal level. One of these indicators is the prevalence of polyclonal infections, or the coexistence of two or more different clones (haplotypes) within the infected individual. As the human parasite genome is haploid, the presence of a single haplotype is expected. The presence of two or more different haplotypes in the sample may be due to the transmission of two or more different genotypes from the same bite (co-infection), or the acquisition of parasites by multiple infections or independent bites (super-infection); and both processes are related to the intensity of transmission. In low transmission there are few infective bites, so there is little chance that a person can acquire a super-infection or co-infection, and therefore most infections are monoclonal. On the other hand, when the necessary conditions exist for the increase in the mosquito population and the interaction between it and the human is favored, super-infections and co-infections increase, generating an increase in the prevalence of polyclonal infections.

The genetic relatedness between parasites is also affected by the intensity of transmission. As we mentioned previously, in low transmission scenarios monoclonal infections are favored, so when an infected person is bitten, a single haplotype is transmitted and self-fertilization of the parasite occurs in the mosquito (sexual reproduction of two genetically identical or related haplotypes). This mechanism of propagation by inbreeding is called clonal propagation, and it causes all the alleles of a haplotype to segregate together and not independently, thus increasing the genetic relatedness in the population.

The parasite effective population size (number of individuals in the population that explains the magnitude of the effect of genetic drift in the population) and its diversity, are also influenced by transmission. The increase in transmission is a consequence of the reproductive success of the different lineages of circulating parasites and the increase in their effective population size. These circumstances increase the introduction of new variants in the population (increase in the population mutation rate or $\theta$), the number of polymorphic sites, the richness of alleles of each polymorphic site (number of alleles per locus), and genetic diversity in terms of heterozygosity of the population (probability of selecting two different alleles by chance). On the other hand, when transmission decreases and the effective population size is reduced, and genetic drift causes the random fixation of alleles, and the diversity of the population is reduced. However, the effective size of the population and its diversity are also strongly influenced by the introduction of new variants as a result of migration and the genetic structure of the parasite population, so the relationship between these metrics and transmission is not linear.

## Complexity of infection and transmission

In this section we will focus on describing the relationship between the intensity of transmission, inferred through the number of samples collected, and the proportion of polyclonal infections. To do this, we will first use the function `fx_get_polygenomic` to identify polyclonal infections, determine their infection complexity (number of clones coexisting in the sample), and the number of heterozygous loci per sample. This information is automatically added to the `metadata` table of the ampseq_object. We will then proceed to describe the distribution of heterozygous loci per sample, the COI, and the proportion of polyclonal infections in each population. Finally, we will describe the temporal change in the proportion of polyclonal infections and its association with the number of samples collected. 

```{r "Defining clonality of the samples"}
ampseq = get_polygenomic(ampseq_object = ampseq, strata = "Subnational_level2", na.rm = FALSE,
                            filters = NULL)

```


```{r 'Distribution of the number of heterozygous loci per sample by sampling location'}

plot_log_NPolyLoci_by_pop = log_scale_histogram(data = ampseq@metadata[ampseq@metadata$NPolyLoci != 0,],
                                                var = "NPolyLoci", binwidth = 1, group_by = "Subnational_level2",
                                                levels = pop_levels, x_label = "Number of heterozygous loci per sample",
                                                fill_color = pop_colors,
                                                y_breaks = c(1, 5,10, 30), ncol = 4)+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 12),
        strip.text = element_text(size = 12)) +
  labs(title = 'B) Distribution of # heterozygous loci per sample', 
       y = "# Samples",
       x = "# heterozygous loci per sample")

```

```{r 'Distribution of COI by sampling location'}

plot_log_coi_by_pop = log_scale_histogram(data = ampseq@metadata,
                                          var = "coi",
                                          binwidth = 1,
                                          group_by = "Subnational_level2",
                                          levels = pop_levels, x_label = "COI",
                                          fill_color = pop_colors,
                                          y_breaks = c(1, 10, 100, 400), ncol = 5)+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 12),
        strip.text = element_text(size = 12)) +
  labs(title = 'C) Distribution of COI by sampling location',
       y = "# Samples")
```

```{r 'Proportion of polyclonal infections by sampling location'}

plot_poly_by_pop = ampseq@pop_summary %>% ggplot(aes(x = factor(pop, levels = c(pop_levels, "Total")),
                                                          y = prop_poly,
                                                          fill = factor(pop, levels = c(pop_levels, "Total"))))+
  geom_col(alpha = .6) +
  geom_errorbar(aes(ymin = prop_poly_lower, ymax = prop_poly_upper), width = .2)+
  theme_bw() +
  labs(title = "D) Frequency of polyclonal infections",
       y = "Frecquency") +
  scale_size(range = c(3, 6)) +
  scale_fill_manual(values = c(pop_colors, "gray30"))+
  scale_y_continuous(limits = c(0, 0.3))+
  theme(axis.text = element_text(size = 12),
        axis.title = element_blank(),
        legend.position = "none")

```

There were `r plot_poly_by_pop$data %>% filter(pop == 'Total') %>% select(n_poly) %>% unlist()` polyclonal infections in the overall population (propottion = `r round(plot_poly_by_pop$data %>% filter(pop == 'Total') %>% select(prop_poly) %>% unlist(),3)`), being Guapi the study area highest number of polyclonals (`r plot_poly_by_pop$data %>% filter(pop == 'Guapi') %>% select(n_poly) %>% unlist()`, proportion = `r round(plot_poly_by_pop$data %>% filter(pop == 'Guapi') %>% select(prop_poly) %>% unlist(), 3)`), followed by Quibdo (`r plot_poly_by_pop$data %>% filter(pop == 'Quibdo') %>% select(n_poly) %>% unlist()`, proportion = `r round(plot_poly_by_pop$data %>% filter(pop == 'Quibdo') %>% select(prop_poly) %>% unlist(), 3)`) and then Buenaventura (`r plot_poly_by_pop$data %>% filter(pop == 'Buenaventura') %>% select(n_poly) %>% unlist()`, proportion = `r round(plot_poly_by_pop$data %>% filter(pop == 'Buenaventura') %>% select(prop_poly) %>% unlist(), 3)`). Moreover in Guapi also showed the largest number of heterozygous loci in a sample (`r plot_log_NPolyLoci_by_pop$data %>% filter(group == 'Guapi', Count > 0) %>% select(bin)%>% max()` heterozygous loci), and was the only one that reported a COI equals to 3 in a sample. However, no significant association was observed between the proportion of polyclonal infections and the number of collected samples. These results suggest that individuals with *P. falciparum* infections in Guapi have a higher exposure to infected mosquitoes than the other two communities, but that the higher exposure is not caused by the increase in transmission intensity in the study area.

```{r 'Temporal change of complexity of infection'}

ampseq@metadata %<>% mutate(Pop_quarter = paste(Subnational_level2, Quarter_of_Collection, sep = '_'))

plot_poly_by_pop_over_time = get_polygenomic(ampseq_object = ampseq, strata = "Pop_quarter", update_popsummary = F, na.rm = TRUE, filters = NULL)[-1,]%>%mutate(
  Population = stringr::str_split(pop, '_', simplify = TRUE)[,1],
  Date = stringr::str_split(pop, '_', simplify = TRUE)[,2],
  prop_poly_lower = case_when(
      prop_poly == 0 ~ 0,
      prop_poly != 0 ~ prop_poly_lower),
      prop_poly_upper = case_when(
        prop_poly == 0 ~ 0,
        prop_poly != 0 ~ prop_poly_upper)
  )%>%
  ggplot(aes(x = Date,
             y = prop_poly,
             ymin = prop_poly_lower,
             ymax = prop_poly_upper,
             fill = factor(Population, levels = pop_levels)))+
  geom_col()+
  geom_errorbar(width = .2)+
  facet_wrap(~factor(Population, levels = pop_levels), ncol = 5)+
  theme_bw()+
  scale_fill_manual(values = pop_colors)+
  labs(title = 'E) Temporal change of the proportion of polyclonal infections',
       y = "Polyclonal infections",
       x = "Date of Collection")+
  theme(axis.text = element_text(size = 10),
        axis.text.x = element_text(angle = 315, vjust = 0),
        axis.title = element_text(size = 12),
        strip.text = element_text(size = 12),
        legend.position =  "none")
```


```{r 'Correlation between the proportion of polyclonal infections and number of samples collected'}
nsamples_vs_proppoly = left_join(plot_poly_by_pop_over_time$data %>% select(pop, prop_poly),
          metadata %>%
  filter(!is.na(Subnational_level2)) %>%
    mutate(Quarter_of_Collection = paste(substr(Date_of_Collection, 1, 4), quarters.Date(Date_of_Collection), sep='-'))%>%
  group_by(Subnational_level2, Quarter_of_Collection) %>%
  summarise(nsamples= n()) %>% mutate(pop = paste(Subnational_level2, Quarter_of_Collection, sep = '_')), by = 'pop')

plot_cor_proppoly_nsamples = ggscatter(nsamples_vs_proppoly, x = "prop_poly", y = "nsamples", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "spearman",
          title = 'F) # Samples vs. Prop. Poly',
          xlab = "Proportion of polyclonal samples", ylab = "Number of samples")

```

```{r 'Complexity of infection and transmission intensity', fig.width=12, fig.height=8, fig.cap = '**Figure 5:** Effect of transmission on the complexity of infections. A) Map os the 3 study areas. B) Distribution of the number of heterozygous loci per samples per population. C) Distribution of the complexity of infection (COI) by study area. In B and C, y-axis is in logarithm scale. D) Proportion of polyclonal infections per study area. E) Proportion of polyclonal infections per study area over year quarters. F) Spearman correlation of the number of collected samples and proportion of polyclonal infections per year quarters. Each point in F represent the measurement of each year quarter in each study area.'}
plot_log_poly_by_pop_map = ggdraw()+
  draw_plot(plot_study_sites,
            x = 0,
            y = .34,
            width = .5,
            height = .66)+
  draw_plot(plot_poly_by_pop,
            x = .5,
            y = .34,
            width = .5,
            height = .22)+
  draw_plot(plot_log_coi_by_pop,
            x = .5,
            y = .56,
            width = .5,
            height = .22)+
  draw_plot(plot_log_NPolyLoci_by_pop,
            x = .5,
            y = .78,
            width = .5,
            height = .22)+
  draw_plot(plot_poly_by_pop_over_time,
            x = 0,
            y = 0,
            width = .67,
            height = .33)+
  draw_plot(plot_cor_proppoly_nsamples,
            x = .67,
            y = 0,
            width = .33,
            height = .33)

plot_log_poly_by_pop_map

```

## Relatedness and Transmission

Relatedness is defined as the probability that, at any locus in the genome, the alleles sampled from two different individuals are identical by descent (IBD). A Hidden Markov Model based method was developed for the estimation of relatedness (HMMIBD), where the relatedness or $r$ is estimated as a function of the haplotype of the two sampled parasites ($Y_i$ and $Y_j$), the frequency of the alleles in the population, the physical distance between loci, the recombination rate ($\rho$), a switching rate of the Markov chain ($k$), and a constant genotyping error rate ($\varepsilon$).

In this section we will measure the pairwise genetic relatedness, and calculate the proportion of pairwise comparisons that have a relatedness $\ge 0.99$ within and between operational units. For that we are going to use the following functions:

1)  `ampseq2loci`: converts the `ampseq_object` into a `loci_object` that only contains the allele information (read abundance is removed). This function only requires the `ampseq_object` as an input argument.

2)  `get_relatedness`: computes pairwise genetic relatedness using the HMMIBD algorithm implemented in `panlejugde`. As an input it requires a `loci_object` and automatically all pairwise comparisons will be computed. Its output is a matrix of all pairwise comparisons. This function is computationally exhaustive, and in future versions a cluster-based parallel version will be integrated. In the current example, the matrix is been generated separately in the cluster at Broadinstitute, and the `.csv` file is loaded by using the function `read.csv`.

3)  `plot_network`: creates a network representation of the pairwise relatedness information. The function requires 7 arguments: 1) the `relatedness_matrix` with all pairwise comparisons, 2) `threshold` to define the minimum prairwise relatedness to plot a branch between the two compared samples, 3) a `metadata` table containing the variables the we want to include in the plot, 4) `sample_id` name of the column in `metadata` table where sample ids are located, 5) `group_by` variable from `metadata` table that we are going to use to assign colors to the nodes in the network, 6) `levels` order of the elements of `group_by` variable, and finally 7) `colors` that are going to be assigned to each element from `group_by`, in this case one color per population.
    
```{r 'generating the loci_object'}
# generating the loci_object----
loci_object = ampseq2loci(ampseq)
```

```{r 'Estimating relatedness'}
# measuring relatedness----

if(file.exists('col_pairwise_relatedness.csv')){
  pairwise_relatedness = as.matrix(read.csv('col_pairwise_relatedness.csv', row.names = 'X'))
}else{
  pairwise_relatedness = NULL
  
  for(w in 1:500){
    start = Sys.time()
    pairwise_relatedness = rbind(pairwise_relatedness,
                                 pairwise_hmmIBD(loci_object, parallel = T, w = w, n = 500))
    time_diff = Sys.time() - start
    
    print(paste0('step ', w, ' done in ', time_diff, ' secs'))
    
  }
  
  write.csv(pairwise_relatedness, 'col_pairwise_relatedness.csv', row.names = F, quote = F)
}
```


```{r 'relatedness distribution within study areas'}

plot_relatedness_distribution_within = plot_relatedness_distribution(
  pairwise_relatedness = pairwise_relatedness,
  metadata = ampseq@metadata,
  Population = 'Subnational_level2',
  fill_color = pop_colors,
  type_pop_comparison = 'within',
  ncol = 5,
  pop_levels = pop_levels
)

```

```{r 'highly related samples within study areas'}
plot_frac_highly_related_within = plot_frac_highly_related(
  pairwise_relatedness = pairwise_relatedness,
  metadata = ampseq@metadata,
  Population = 'Subnational_level2',
  fill_color = pop_colors,
  threshold = 0.99, type_pop_comparison = 'within', pop_levels = pop_levels)

```

```{r 'fraction of highly related samples over time (3 months intervals) within study areas'}
plot_frac_highly_related_over_quarters_within = plot_frac_highly_related_over_time(pairwise_relatedness = pairwise_relatedness,
                                       metadata = ampseq@metadata,
                                       Population = c('Subnational_level2', 'Month_of_Collection'),
                                       fill_color = pop_colors,
                                       threshold = 0.99,
                                       type_pop_comparison = 'within',
                                       pop_levels = pop_levels, ncol = 1)
```


```{r 'Correlation between the proportion of highly related samples and number of samples collected'}
nsamples_vs_highlyR = left_join(plot_frac_highly_related_over_quarters_within$plot_frac_highly_related$data %>% mutate(pop = paste(Pop_comparison, Date_Yi, sep = '_'))%>%
                                   ungroup()%>%
                                   select(pop, prop),
          metadata %>%
  filter(!is.na(Subnational_level2)) %>%
  group_by(Subnational_level2, Month_of_Collection) %>%
  summarise(nsamples= n()) %>% mutate(pop = paste(Subnational_level2, Month_of_Collection, sep = '_')), by = 'pop')  %>%
  filter(prop != 0, Subnational_level2 %in% c('Quibdó', 'Buenaventura', 'Guapi'))

plot_cor_highlyR_nsamples = ggscatter(nsamples_vs_highlyR, x = "prop", y = "nsamples", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "spearman",
          title= 'D) # Collected samples vs. Proportion of highly related samples',
          xlab = "Proportion of highly related samples", ylab = "Number of samples",
          facet.by = 'Subnational_level2')

```

There were `r length(pairwise_relatedness[!is.na(pairwise_relatedness)]) -nrow(pairwise_relatedness)` pairwise comparisons, and as it can be observed in figure 6, most samples were highly related (r $\ge$ 0.99) and the average genetic relatedness was `r round(mean(pairwise_relatedness, na.rm = T), 3)` (se = `r round(sd(pairwise_relatedness, na.rm = T)/sqrt(nrow(pairwise_relatedness)), 3)`). The average pairwise relatedness and the proportion of highly related sample comparisons within Buenaventura (mean $r$ = `r round(plot_relatedness_distribution_within$relatedness %>% filter(Pop_comparison == 'Buenaventura') %>% summarise(mean(r)) %>% unlist(), 3)`, proportion = `r plot_frac_highly_related_within$highly_related_table %>% ungroup %>% filter(Pop_comparison == 'Buenaventura') %>% select(prop) %>% unlist`) were greater than the other two study areas (`r round(plot_relatedness_distribution_within$relatedness %>% filter(Pop_comparison %in% c('Guapi','Quibdo')) %>% summarise(mean(r)) %>% unlist(), 3)`, proportion = `r round(plot_frac_highly_related_within$highly_related_table %>% ungroup %>% filter(Pop_comparison %in% c('Guapi','Quibdo')) %>% select(prop) %>% unlist %>% mean, 3)`). These results suggest that the genetic diversity of the parasite population or the transmission intensity in Buenventura is lower compared to the other sites. On the other hand, an strong negative correlation between the temporal change of the proportion of highly related samples and the number of collected samples was observed in Quibdo but not in the other two study areas (Figure 7 C and D), suggesting the increase of transmission intensity (resent outbreak) in this site.

```{r 'network', fig.width=5, fig.height=5, fig.cap = '**Figure 6:** Network representation of pairwise genetic relatedness between samples. Each node represents a sample, and branches are displayed only for pairwise comparisons with a relatedness greater than 0.99. Colors were assigned based on the sampling location: Buenaventura (blue), Guapi (glold) and Quibdo (red).'}

source('fx_plot_network.R')

plot_network = plot_network(pairwise_relatedness = pairwise_relatedness,
                               threshold = .95,
                               metadata = ampseq@metadata,
                               sample_id = 'Sample_id',
                               group_by = 'Subnational_level2',
                               levels = pop_levels,
                               colors = pop_colors
                              )

```

```{r, fig.width = 12, fig.height=10, fig.cap = '**Figure 7:** Effect of transmission on relatedness. A) Histogram of the distribution of pairwise genetic relatedness by study area (Panels). The dashed vertical line represents the average genetic relatedness in the overall population. B) Proportion of highly related samples (IBD > 0.99) in each study area. C) Proportion of highly related samples over year quarters in each study area. D) Spearman correlation of the proportion of highly related samples per year quarter in each estudy area and the number of collected samples by passive case detection.'}

plot_highlyR_over_time = ggdraw()+
  draw_plot(plot_relatedness_distribution_within$plot+
              labs(title = 'A) Distribution of relatedness by study area'),
            x = 0,
            y = .66,
            width = .66,
            height = .34)+
  draw_plot(plot_frac_highly_related_within$plot+
              labs(title = 'B) Highly related',
                   y = 'Proportion'),
            x = .66,
            y = .66,
            width = .34,
            height = .34)+
  draw_plot(plot_frac_highly_related_over_quarters_within$plot_frac_highly_related+
              labs(title = 'C) Proportion of highly related samples over time (IBD > 0.99)',
                   y = 'Proportion'),
            x = 0,
            y = .33,
            width = 1,
            height = .33)+
  draw_plot(plot_cor_highlyR_nsamples,
            x = 0,
            y = 0,
            width = 1,
            height = .33)

plot_highlyR_over_time

```


```{r 'getting genetic diversity by loci', include = FALSE}

## Haplotypic and genetic diversity and transmission
source("fx_loci_diversity.R")

ampseq$loci_performance = cbind(ampseq$loci_performance,fx_loci_diversity(ampseq$loci_abd_table, variance = TRUE))
```


```{r 'getting genetic diversity by pop', include = FALSE}

source("fx_pop_diversity.R")

ampseq[["pop_summary"]] = cbind(ampseq[["pop_summary"]],fx_pop_diversity(ampseq_object = ampseq, strata = "Population")[-1])

```

```{r 'Haplotype and genetic diversity by sampling location', fig.width=12, fig.height=6, include = FALSE}

plot_div_by_pop = ampseq[["pop_summary"]]%>%
  pivot_longer(cols= all_of(c("Effective.richness",
                              "Shannon",
                              "Hexp")),
               names_to = "metric", values_to = "value")%>%
  ggplot(aes(x = factor(pop, levels = c(pop_levels, "Total")),
             y = value,
             color = factor(pop, levels = c(pop_levels, "Total")),
             size = Richness))+
  geom_point(alpha = .6) +
  facet_wrap(~factor(metric,
                     levels = c("Effective.richness",
                                "Shannon",
                                "Hexp")),
             scales = "free_y",
             ncol = 1) +
  theme_bw() +
  scale_color_manual(values = c(pop_colors, "gray30"))+
  labs(title = "Haplotypic and genetic diversity",
       y = "Value",
       x = "Population",
       color = "Population")+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 12),
        strip.text = element_text(size = 12))


plot_div_by_pop_map = ggdraw()+
  draw_plot(plot_study_sites,
            x = 0,
            y = .1,
            width = .4,
            height = .8)+
  draw_plot(plot_div_by_pop,
            x = .4,
            y = 0,
            width = .6,
            height = 1)

plot_div_by_pop_map

# ggsave("plot_div_by_pop_map.png",
#        plot_div_by_pop_map,
#        device = "png",
#        units = "in",
#        width = 12,
#        height = 6,
#        dpi = 320)
```

# Measuring geographic connectivity

Malaria transmission is heterogeneous throughout the Colombian territory, and it is mainly concentrated in transmission foci where demographic and environmental factors favor the interaction between the two hosts of the parasite, humans and Anopheles mosquitoes. Operationally, disease control units are defined based on sub-national administrative divisions. The connectivity or dispersal pattern of the parasite between these geographic units may limit progress in disease control if activities are not properly coordinated. This connectivity is mainly influenced by human movement, and can be inferred by estimating the proportion of genetically closely related parasites between the different areas.

1)  `fx_plot_relatedness_distribution`: visualizes the distribution of the pairwise relatedness within and between the different study areas. Its arguments are `relatedness_matrix`, `metadata`, `Population`, and `fill_color`.

2)  `fx_frac_highly_related`: visualizes the fraction of highly related haplotypes within and between study areas. It has the same arguments than the previous function plus `threshold` (minimum value to consider two samples to be highly related).

```{r 'relatedness distribution between study areas'}

plot_relatedness_distribution_between = fx_plot_relatedness_distribution(
  relatedness_matrix = pairwise_relatedness,
  metadata = ampseq$metadata,
  Population = 'Population',
  fill_color = c("gray50", "gray50", "gray50", "gray50", "gray50", "gray50"),
  type_pop_comparison = 'between',
  ncol = 3,
  pop_levels = c("Buenaventura-Guapi", "Buenaventura-Quibdo", "Buenaventura-Tumaco",
                 "Guapi-Quibdo", "Guapi-Tumaco", "Quibdo-Tumaco")
)
```

```{r 'highly related samples between study areas'}

plot_frac_highly_related_between = fx_plot_frac_highly_related(
  relatedness_matrix = pairwise_relatedness,
  metadata = ampseq$metadata,
  Population = 'Population',
  fill_color = c("gray50", "gray50", "gray50","gray50", "gray50", "gray50"),
  threshold = 0.99, type_pop_comparison = 'between', pop_levels = c("Buenaventura-Guapi", "Buenaventura-Quibdo", "Buenaventura-Tumaco",
                 "Guapi-Quibdo", "Guapi-Tumaco", "Quibdo-Tumaco"))
```

The average genetic relatedness and the proportion of highly related samples were higher between Buenaventura and Guapi (mean $r$ = `r round(plot_relatedness_distribution_between$relatedness %>% filter(Pop_comparison == 'Buenaventura-Guapi') %>% summarise(mean(r)) %>% unlist(), 3)`, proportion = `r round(plot_frac_highly_related_between$highly_related_table %>% ungroup %>% filter(Pop_comparison == 'Buenaventura-Guapi') %>% select(prop) %>% unlist %>% mean, 3)`) than Buenaventura and Quibdo (mean $r$ = `r round(plot_relatedness_distribution_between$relatedness %>% filter(Pop_comparison == 'Buenaventura-Quibdo') %>% summarise(mean(r)) %>% unlist(), 3)`, proportion = `r round(plot_frac_highly_related_between$highly_related_table %>% ungroup %>% filter(Pop_comparison == 'Buenaventura-Quibdo') %>% select(prop) %>% unlist %>% mean, 3)`), and Guapi and Quibdo (mean $r$ = `r round(plot_relatedness_distribution_between$relatedness %>% filter(Pop_comparison == 'Guapi-Quibdo') %>% summarise(mean(r)) %>% unlist(), 3)`, proportion = `r round(plot_frac_highly_related_between$highly_related_table %>% ungroup %>% filter(Pop_comparison == 'Guapi-Quibdo') %>% select(prop) %>% unlist %>% mean, 3)`). The results suggest higher connectivity or mobilization  of parasites between Buenaventura and Guapi than Quibdo and these two study areas.

```{r 'Connectivity', fig.width=10, fig.height=5, fig.cap = '**Figure 8:** A) Histogram distribution of pairwise genetic relatedness between sampling locations. Panels in columns correspond to the three study areas. The dotted vertical lines represent the average genetic relatedness in the whole population. B) Proportion of highly related samples between sampling locations.'}

plot_connectivity = ggdraw()+
  draw_plot(plot_relatedness_distribution_between$plot+
              labs(title = 'A) Distribution of relatedness'),
            x = 0,
            y = 0,
            width = .7,
            height = 1)+
  draw_plot(plot_frac_highly_related_between$plot+
              labs(title = 'B) Highly related'),
            x = 0.7,
            y = 0,
            width = .3,
            height = 1)

plot_connectivity
```

# Preliminary conclusions

1. The haplotype associated to sulfadoxine resistance is differentially spread across the study areas.

2. Individuals with *P. falciparum* infections in Guapi have a higher exposure to infected mosquitoes than the other two communities.

3. The genetic diversity of the parasite population or the transmission intensity in Buenventura is lower compared to the other two sites.

4. There is an increment of transmission intensity (resent outbreak) in Quibdo.

5. There is higher connectivity or mobilization  of parasites between Buenaventura and Guapi.

