---
title: "Micro-Haplotype (MHap) Analysis"
author: "Neafsey Lab"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

The purpose of this repository is to introduce researchers in Neafsey Lab in the analysis of genotyping information generated through amplicon sequencing (also called Microhaplotype genotyping).

This covers 4 tutorials:

- [Intro to R](),
- [Basic Bash](),
- The analysis of the [AMPLseq]() data for *Plasmodium falciparum*, and
- The analysis of the [*PvGTSeq*]() data for *Plasmodium vivax*.

Our GitHub repository contains a set of functions that allows the user to upload and manipulate the genotypic information stored in .tsv tables (in CIGAR format) that has been generated by the [malaria-amplicon-pipeline](https://github.com/broadinstitute/malaria-amplicon-pipeline) of our lab. All those functions and the required libraries are in the R scripts [amplseq_functions.R]() and [amplseq_required_libraries.R](), and an additional set of Rcpp functions for the estimation of IBD and eigenvectors are in the files [hmmloglikelihood.cpp]() and [Rcpp_functions.cpp](). Some of the functions requires additional input files such as fasta files for reference strains or tables of alleles associated with drug resistance. These additional information is stored in the folder [`doc/reference`]().

The [AMPLseq]() and [*PvGTSeq*]() tutorials have been created to introduce the final users to all the set of functions in this repository. To run this tutorial a data sets are stored in the folder [`doc/data`](). The samples for these tutorials comes from a collaboration with the groups of Caucaseco in Colombia, Stella Chenet in Peru, and David Forero in Venezuela.

