# Micro-Haplotype (MHap) Analysis

The purpose of this repository is to introduce researchers in Neafsey
Lab in the analysis of genotyping information generated through amplicon
sequencing (also called Microhaplotype genotyping).

This covers 8 tutorials:

-   [Intro to MHap-Analysis pipeline](https://paulonvnv.github.io/MHap-Analysis/MHap_Analysis_pipeline.html),

-   [Performance of amplicon sequencing data](https://paulonvnv.github.io/MHap-Analysis/Sequencing_Performance.html),

-   [Drug Resistance Surveillance](https://paulonvnv.github.io/MHap-Analysis/Drug_resistance_surveillance.html),

-   [Complexity of Infection](https://paulonvnv.github.io/MHap-Analysis/Complexity_of_infection.html)

-   An example of a customized analysis of the
    [AMPLseq](https://paulonvnv.github.io/MHap-Analysis/MHapPfalciparum.html)
    data for *Plasmodium falciparum*, and

-   [*PvGTSeq*](https://paulonvnv.github.io/MHap-Analysis/MHapPvivax.html) data for *Plasmodium vivax*.

-   [Intro to R](https://paulonvnv.github.io/MHap-Analysis/Intro_to_R.html),

-   [Basic Bash](https://paulonvnv.github.io/MHap-Analysis/basic_bash.html),

Our GitHub repository contains a set of functions that allows the user
to upload and manipulate the genotypic information stored in .tsv tables
(in CIGAR format) that has been generated by the
[malaria-amplicon-pipeline](https://github.com/broadinstitute/malaria-amplicon-pipeline)
of our lab. All those functions and the required libraries are in the R
scripts
[amplseq_functions.R](https://github.com/Paulonvnv/MHap-Analysis/blob/main/docs/functions_and_libraries/amplseq_functions.R)
and
[amplseq_required_libraries.R](https://github.com/Paulonvnv/MHap-Analysis/blob/main/docs/functions_and_libraries/amplseq_required_libraries.R),
and an additional set of Rcpp functions for the estimation of IBD and
eigenvectors are in the files
[hmmloglikelihood.cpp](https://github.com/Paulonvnv/MHap-Analysis/blob/main/docs/functions_and_libraries/hmmloglikelihood.cpp)
and
[Rcpp_functions.cpp](https://github.com/Paulonvnv/MHap-Analysis/blob/main/docs/functions_and_libraries/Rcpp_functions.cpp).
Some of the functions require additional input files such as fasta files
for reference strains or tables of alleles associated with drug
resistance. These additional information is stored in the folder
[`doc/reference`](https://github.com/Paulonvnv/MHap-Analysis/tree/main/docs/reference).

The
[AMPLseq](https://paulonvnv.github.io/MHap-Analysis/MHapPfalciparum.html)
and
[*PvGTSeq*](https://paulonvnv.github.io/MHap-Analysis/MHapPvivax.html)
tutorials have been created to introduce the final users to all the set
of functions in this repository, while the [Intro to MHap-Analysis
pipeline](https://paulonvnv.github.io/MHap-Analysis/MHap_Analysis_pipeline.html)
tutorial shows how to generate automated reports. To run these tutorials
data sets are stored in the folder
[`doc/data`](https://github.com/Paulonvnv/MHap-Analysis/tree/main/docs/data).
The samples for these tutorials comes from a collaboration with the
groups of Caucaseco in Colombia, Stella Chenet in Peru, and David Forero
in Venezuela.

All tutorials are [here](https://paulonvnv.github.io/MHap-Analysis/).



